metpy/calc/basic.py

```diff
@@ -106,6 +107,12 @@ def get_wind_components(speed, wdir):
      <Quantity(7.071067811865477, 'meter / second')>)
 
     """
+    try:
+        wdir = wdir.to('radians')
```

Comment:
If you're doing this, you could just drop units at this point--wouldn't need units later then.